app.py

```diff
@@ -252,62 +240,64 @@ def internal_server_error(e):
 def handle_postback():
     today_date = datetime.now().strftime("%m/%d/%Y %H:%M")
     citybase_data = request.get_json()
+    # TODO: validate citybase_data
+    # information from citybase payload
     app.logger.info(f"New POST with payload: {citybase_data}")
-    knack_record_id = get_knack_record_id(citybase_data)
-    # if knack_record_id is not a string, then it is an error tuple, for example ("Missing custom attributes", 400)
-    if not isinstance(knack_record_id, str):
-        # returns an error message and error code
-        return knack_record_id
-
-    knack_invoice = get_knack_invoice(citybase_data)
-    # if knack_invoice is not a string, then it is an error tuple
-    if not isinstance(knack_invoice, str):
-        # returns an error message and error code
-        return knack_invoice
-
+    custom_attributes = unpack_custom_attributes(citybase_data["data"]["custom_attributes"])
+    knack_record_id = custom_attributes["knack_record_id"]
```

Comment:
if you want to avoid key errors, `custom_attributes.get("popertry")` might be safer here? or wrapping the `citybase_data` parsing in a `try/except` could serve as  quick 'n dirty validation engine...